ethcore/src/engines/tendermint/mod.rs

```diff
@@ -491,14 +497,15 @@ impl Engine for Tendermint {
 	fn verify_block_basic(&self, header: &Header, _block: Option<&[u8]>) -> Result<(), Error> {
 		let seal_length = header.seal().len();
 		if seal_length == self.seal_fields() {
+			let proposal_len = header.seal()[1].len();
 			let signatures_len = header.seal()[2].len();
-			if signatures_len >= 1 {
+			if (proposal_len == 1) ^ (signatures_len == 1) {
```

Comment:
This is sort of wonky, since its just the length of the seal field (1 means no signatures). Will change it to comparing actual RLPs.